src/main/java/bio/terra/workspace/app/controller/ControlledGoogleResourceApiController.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.workspace.app.controller;
+
+import bio.terra.workspace.common.utils.ControllerValidationUtils;
+import bio.terra.workspace.generated.controller.ControlledGoogleResourceApi;
+import bio.terra.workspace.generated.model.CreateControlledGoogleBucketRequestBody;
+import bio.terra.workspace.generated.model.CreatedControlledGoogleBucket;
+import bio.terra.workspace.service.controlledresource.ControlledGoogleResourceService;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequestFactory;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.job.JobService.AsyncJobResult;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public class ControlledGoogleResourceApiController implements ControlledGoogleResourceApi {
+  private final Logger logger =
+      LoggerFactory.getLogger(ControlledGoogleResourceApiController.class);
+
+  private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  private final ControlledGoogleResourceService controlledResourceService;
+  private final HttpServletRequest request;
+  private final JobService jobService;
+
+  @Autowired
+  public ControlledGoogleResourceApiController(
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+      ControlledGoogleResourceService controlledResourceService,
+      JobService jobService) {
+    this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+    this.controlledResourceService = controlledResourceService;
+    this.request = getRequest().orElse(null);
```

Comment:
I'm not sure about this. I noticed IJ complaining something wasn't Kosher about injecting the request and saw this method in the parent class.

Reply:
![Screen Shot 2021-02-11 at 9 38 50 AM](https://user-images.githubusercontent.com/53479492/107651159-30af2b80-6c4d-11eb-99cc-f2564d349bba.png)

Reply:
switched back to injecting.